superset/alfred/utils.py

```diff
@@ -0,0 +1,354 @@
+                )
+                if dept not in dept_map:
+                    dept_map[dept] = event_count
+                    event_count += 1
+
+                # Set the description per event.
+                # If logs supply 'Event Info' it will be grabbed from all the logs, uniqued and sorted into the description
+                if dept not in info_map:
+                    info_map[dept] = "\n".join(
+                        sorted(
+                            list(
+                                set(
+                                    [
+                                        d["Event Info"]
+                                        for d in data_set
+                                        if "Event Info" in d
+                                        and (
+                                            (
+                                                dept != "Unknown Department"
+                                                and "Event Organization" in d
+                                                and dept == d["Event Organization"]
+                                            )
+                                            or (
+                                                dept == "Unknown Department"
+                                                and "Event Organization" not in d
+                                            )
+                                        )
+                                    ]
+                                )
+                            )
+                        )
+                    )
+                    log["event description"] = (
+                        "{" + markup + "}" + f"{dept}\n{info_map[dept]}".strip()
+                    )
+                elif dept in info_map:
+                    log["event description"] = (
+                        "{" + markup + "}" + f"{dept}\n{info_map[dept]}".strip()
+                    )
+                else:
+                    log["event description"] = "{" + markup + "}" + f"{dept}".strip()
+
+                # Set the Event ID
+                log["new event id"] = f"LS-<{event_type}_{dept_map[dept]}>"
+
+    # Create Retention and Stage it
+    logger.info("Creating Retention...")
+
+    # These lines were copied from the stage function in the RetentionInstance
+    # class in the alfred_client library, but does not block and wait for the
+    # retention to be in the 'ready' state. This async case will hopefully be 
+    # added to the library in the future and this code can be replaced.
+    out_data_set: List[DataSetEntry] = DataSetMapper().map_data_set(data_set)
+    retention = alfred.create_retention(
+            f"Harmonized Email Dashboard Retention - {datetime.now()}"
+        )
+    retention = alfred.add_retention_dataset(
+            retention.uri,
+            out_data_set,
+            metadata_system,
+            mapping_name="hogwarts.harmonized.eml_metadata"
+        )
```

Comment:
these are the lines that I added to return the retention url before it is in the ready state

Reply:
Are we good with that name, _Harmonized Email Dashboard Retention - Timestamp_? Technically, any dataset that has the advanced type can retain to Alfred, not necessary from the dashboard.